### PR TITLE
Add recipe to run script/shoryuken

### PR DESCRIPTION
**Pull request by ktheory**
The config was copied from the delayed_job recipe.

We want to run shoryuken as part of #35.

## Prerequisites

- [x] deploy rubygems/rubygems.org#1176, which adds `bin/shoryuken-worker`. This will fail otherwise.

### CODE REVIEW

File: cookbooks/rubygems-app/recipes/logrotate.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@
   options %w(missingok nocreate compress delaycompress dateext notifempty sharedscripts)
   postrotate [
     '    [ -f /etc/service/unicorn/supervise/pid ] && kill -USR1 `cat /etc/service/unicorn/supervise/pid`',
-    '    [ -f /etc/service/delayed_job/supervise/pid ] && kill -USR1 `cat /etc/service/delayed_job/supervise/pid`'
+    '    [ -f /etc/service/delayed_job/supervise/pid ] && kill -USR1 `cat /etc/service/delayed_job/supervise/pid`',
+    '    [ -f /etc/service/shoryuken/supervise/pid ] && kill -USR1 `cat /etc/service/shoryuken/supervise/pid`'
```
Comment (dwradcliffe) [inline cookbooks/rubygems-app/recipes/logrotate.rb]: Does shoryuken handle USR1 for reopening log files?

Comment (ktheory) [inline cookbooks/rubygems-app/recipes/logrotate.rb]: @dwradcliffe Glad you asked!

`USR1` is actually shoryuken's graceful shutdown signal. It doesn't have a "reopen logs" signal like unicorn, nginx, etc. So this works works in the sense that shoryuken will restart, hence opening new logs. But it's not obvious. I'll update to just do a more obvious `/etc/init.d/shoryuken restart`.

See https://github.com/phstc/shoryuken/wiki/Signals for more info.

Comment (dwradcliffe) [inline cookbooks/rubygems-app/recipes/logrotate.rb]: Awesome. I asked because the delayed job one wasn't working. :)

Comment (ktheory) [inline cookbooks/rubygems-app/recipes/logrotate.rb]: Sounds like a fun thing to debug. :laughing:

I updated the PR so delayed_job and shoryuken use similar restart commands.

File: cookbooks/rubygems-app/recipes/logrotate.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@
   options %w(missingok nocreate compress delaycompress dateext notifempty sharedscripts)
   postrotate [
     '    [ -f /etc/service/unicorn/supervise/pid ] && kill -USR1 `cat /etc/service/unicorn/supervise/pid`',
-    '    [ -f /etc/service/delayed_job/supervise/pid ] && sv -w 30 restart delayed_job'
+    '    [ -f /etc/service/delayed_job/supervise/pid ] && sv -w 30 restart delayed_job',
+    '    [ -f /etc/service/delayed_job/supervise/pid ] && sv -w 30 restart shoryuken'
```
Comment (dwradcliffe) [inline cookbooks/rubygems-app/recipes/logrotate.rb]: wrong pid path here

Comment (ktheory) [inline cookbooks/rubygems-app/recipes/logrotate.rb]: D'oh! Fixed. :bow:

**Pull request merged**